### PR TITLE
Bjorncs/pom file validation

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/application/validation/BundleValidator.java [hunk 1]
```diff
@@ -117,27 +133,91 @@ private static void validateImportedPackages(DeployLogger deployLogger, String f
...
+        } catch (ParserConfigurationException e) {
+            throw new RuntimeException(e);
+        } catch (Exception e) {
+            deployLogger.log(Level.INFO, String.format("Unable to parse pom.xml from %s", jarFilename));
+        }
+    }
+
+    private enum DeprecatedMavenArtifact {
+        VESPA_HTTP_CLIENT_EXTENSION("com.yahoo.vespa", "vespa-http-client-extensions",
+                "The 'vespa-http-client-extensions' artifact will be removed in Vespa 8. " +
```
Comment (gjoranv) [inline config-model/src/main/java/com/yahoo/vespa/model/application/validation/BundleValidator.java:183]: It seems unnecessary to repeat the artifact name? Couldn't we just say "This artifact"?

**Review by gjoranv — APPROVED** [review]
🎖️

File: config-model/src/main/java/com/yahoo/vespa/model/application/validation/BundleValidator.java [hunk 2]
```diff
@@ -117,27 +133,91 @@ private static void validateImportedPackages(DeployLogger deployLogger, String f
...
+                    .compile("/project/dependencies/dependency")
+                    .evaluate(pom, XPathConstants.NODESET);
+            for (int i = 0; i < dependencies.getLength(); i++) {
+                Element dependency = (Element) dependencies.item(i);
+                String groupId = dependency.getElementsByTagName("groupId").item(0).getTextContent();
+                String artifactId = dependency.getElementsByTagName("artifactId").item(0).getTextContent();
+                for (DeprecatedMavenArtifact deprecatedArtifact : DeprecatedMavenArtifact.values()) {
+                    if (groupId.equals(deprecatedArtifact.groupId) && artifactId.equals(deprecatedArtifact.artifactId)) {
+                        deployLogger.logApplicationPackage(Level.WARNING,
+                                String.format("For pom.xml in '%s': \n" +
```
[the remark this answers is not in the document]
Comment (bjorncs) [reply config-model/src/main/java/com/yahoo/vespa/model/application/validation/BundleValidator.java]: Fixed

File: config-model/src/main/java/com/yahoo/vespa/model/application/validation/BundleValidator.java (hunk 1 above)
Comment (bjorncs) [reply config-model/src/main/java/com/yahoo/vespa/model/application/validation/BundleValidator.java]: Fixed

**Pull request merged**